apps/prairielearn/assets/scripts/lib/tabulator.ts

```diff
@@ -0,0 +1,134 @@
+import { parseHTMLElement } from '@prairielearn/browser-utils';
+import { html } from '@prairielearn/html';
+import { v4 as uuid } from 'uuid';
+import {
+  Tabulator,
+  FormatModule,
+  EditModule,
+  FilterModule,
+  SortModule,
+  PageModule,
+  MutatorModule,
+  FrozenColumnsModule,
+  ResizeTableModule,
+  Options,
+} from 'tabulator-tables';
+
+Tabulator.registerModule([
+  FormatModule, // custom formatters
+  EditModule, // editors, also required for header filters
+  FilterModule, // column header filters
+  SortModule, // sorting by clicking on headers
+  PageModule, // pagination
+  MutatorModule, // calculate data derived from other columns
+  FrozenColumnsModule, // freeze first column
+  ResizeTableModule, // recalculate column widths on page resize
+]);
+
+type DefaultTabulatorOptions = Options & {
+  columnVisibilityDropdown?: Element | null;
+};
+
+export function defaultTabulator(
+  elementQueryString: string,
+  options: DefaultTabulatorOptions,
+): Tabulator {
+  const table: Tabulator = new Tabulator(elementQueryString, {
+    layout: 'fitData',
+    pagination: true,
+    paginationCounter: (pageSize, currentRow, _currentPage, totalRows) =>
+      `Showing ${currentRow}-${currentRow + pageSize - 1} of ${totalRows} question${
+        totalRows === 1 ? '' : 's'
+      }` +
+      (totalRows === table.getData().length
+        ? ''
+        : ` (filtered from ${table.getData().length} total question${
+            table.getData().length === 1 ? '' : 's'
+          })`),
+    paginationSize: 50,
+    paginationSizeSelector: [10, 20, 50, 100, 200, 500, true],
```

Comment:
When drawing the pagination controls, would it be possible to use icons rather than "first", "next", etc? I found that I had to do quite a lot of reading/thinking to figure out what to click. I think that https://icons.getbootstrap.com/icons/chevron-bar-left/ and https://icons.getbootstrap.com/icons/chevron-left/ would be clearer.